### PR TITLE
Split hi res geocoding

### PR DESCRIPTION
**Pull request by rafatower**
Just to get an idea of how many tests I broke

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4694/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4695/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4733/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4737/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4746/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4760/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4761/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4767/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4777/))

**Comment by rafatower** [comment]
What's here:
- Split the former `Geocoder` (here) into `HiresGeocoder` and `HiresBatchGeocoder`: now they know about the service and API they work with but not much about db's, tables or the rest.
- A new class `HiresGeocoderFactory` to decide whether to instantiate one or another.
- Many changes to the `TableGeocoder` to accommodate the above. E.g: moved the state polling loop on the here service to the `HiresBatchGeocoder`, so from the client classes that's kind of transparent.
- Stuff from public to protected/private
- Fixes for legacy tests, now turned into kind of white box (accessing their privates :stuck_out_tongue_winking_eye: )
- Many small details, bugfixes and some TODO's.

Some comments about this:
- I'd like to add more tests, they're easier now but this task will have to be "time boxed" so I'll do my best.
- I have not thrown away the existing tests yet because they proved to be useful to check against expected behavior, though ideally they're to be deleted.
- Please don't be too hard on it, I don't like remaining TODO's there but again I tried to carry on by priority and the TODO's let me capture things in context that otherwise are lost in wiki pages/tickets...

@Kartones and / or @juanignaciosl please review

cc/ @ethervoid so that you can get familiar with the code. Don't get too scared :smile:

**Comment by juanignaciosl** [comment]
LGTM :+1: :clap:

File: services/geocoder/lib/hires_batch_geocoder.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        sleep polling_sleep_time
+        update_status
+      end
+
+      # TODO: do here all the process_results stuff
+
+    end
+
+    def upload
+      assert_batch_api_enabled
```
Comment (Kartones) [inline services/geocoder/lib/hires_batch_geocoder.rb]: As we discussed in person and you said would be better at the factory, but at least a good starting point!

**Comment by Kartones** [comment]
:+1:

**Pull request merged**